lib/Remotery.c

```diff
@@ -1705,7 +1708,12 @@ static rmtThreadId rmtGetCurrentThreadId()
 #ifdef RMT_PLATFORM_WINDOWS
     return GetCurrentThreadId();
 #else
-    return 0;
+#elif defined(RMT_PLATFORM_POSIX)
```

Comment:
I've been looking at gettid, syscall with __NR_gettid/SYS_gettid, and pthread_threadid_np. They seem to be a mess of incompatibilities across all the supported platforms. Could not this function be implemented in terms of pthread_self for all the POSIX platforms? It at least is already compiling and being used on the supported platforms. My thoughts are similar to this: http://elliotth.blogspot.com/2012/04/gettid-on-mac-os.html

Reply:
My thinking here was that pthread_self returned a "pthread_t" which is an unsigned long int,  which might not fit into "rmtThreadId" which is a "unsigned int".
On the other hand, "syscall" also returns a "long int", so that might also be true there.
Perhaps better to use pthread_self on *NIX and then make rmtThreadId into a "rmtU64" just to be on the safe side?

Reply:
Yeah, I think making `rmtThreadId` a `rmtU64` on all platforms would be the safest option.